Update to split sampler and scheduler selections

rhyswynn opened the pull request:
This will add an additional scheduler drop down selector, and pass the additional scheduler parameter to the SD pipeline, based on recent changes to the Automatic1111 approach.

rhyswynn committed:
Update to split sampler and scheduler selections

andyxr commented:
What happens when users try to load settings files from an earlier version of Deforum? Is this case handled?

rhyswynn commented:
I'm happy to push these changes to a branch instead, or you could test it from my fork.

rhyswynn commented:
Good point, I think it would result in the default values, if the keys don't match it won't have a value to update. I'll review the logic later today. I am confident it is a non-breaking concern, but it would be a more complete solution that way.

rhyswynn committed:
Enabled legacy sampler settings to be imported

rhyswynn commented:
I updated it. defaults.py I added an additional LCM sampler that's showing in A1111 txt to image. I have no idea how that logic in settings.py was working before, because it was checking if a string value was an int, so the logic to set the sampler values from the settings file couldn't have been working. I added logic to parse the value and set the scheduler value if the legacy value was a combination. I tested it out and it successfully loaded the old settings with the correct sampler/scheduler combination.

rhyswynn committed:
Enabled legacy sampler settings to be imported

rhyswynn commented:
this needs more work, don't approve it yet please

rhyswynn committed:
Enabled legacy sampler settings to be imported

rhyswynn commented:
Ok, all good now, thank you for your patience! Python isn't my native language :)

andyxr commented:
> Ok, all good now, thank you for your patience! Python isn't my native language :)

Ditto!! :-)

andyxr commented:
Can this be merged now?

rhyswynn commented:
Yep, it's working fine for me with the updates I made. Thank you!

rhyswynn commented:
I don't think it would work, because the sd pipeline in the older version is only expecting the sampler parameter, not the sampler and the scheduler. That's the never ending challenge with these kind of add ons to another package. How much time and effort do we spend to accommodate the edge case of someone who's had A1111 for a while and suddenly wants to use Deforum? It wouldn't be technically challenging to add a check, but I don't have the time or the motivation for doing that and doing a downgrade to test it. I vote to add a note to the readme. Let me know your thoughts and I'll update the pull request.

andyxr commented:
> I don't think it would work, because the sd pipeline in the older version is only expecting the sampler parameter, not the sampler and the scheduler. That's the never ending challenge with these kind of add ons to another package. How much time and effort do we spend to accommodate the edge case of someone who's had A1111 for a while and suddenly wants to use Deforum? It wouldn't be technically challenging to add a check, but I don't have the time or the motivation for doing that and doing a downgrade to test it. I vote to add a note to the readme. Let me know your thoughts and I'll update the pull request.

Yeah, I get it. Like I said, we had to force everyone to use A1111 v1.8, so why not 1.9. Let's merge it and release.

andyxr commented:
Shall we merge this puppy?

rhyswynn commented:
> Shall we merge this puppy?

Yes please. Do I need to take some action? I'm sorry, I have not contributed to someone else's repo before.

rewbs commented:
> Do I need to take some action? I'm sorry, I have not contributed to someone else's repo before.

Nothing more needed from you @rhyswynn – I am reviewing now and it looks good. I will merge shortly.

>  I think we can live with it being a V1.9 only thing

Agreed. I have created a release with the current state of the repo (before merging this) tagged as the version to use for a1111 v1.8:  https://github.com/deforum-art/sd-webui-deforum/releases/tag/a1111-v1.8
